modules/ovn-kubernetes-logical-architecture-con.adoc

```diff
@@ -0,0 +1,68 @@
+
+Gateway routers:: Gateway routers sometimes called L3 gateway routers, are typically used between the distributed routers and the physical network. Gateway routers including their logical patch ports are bound to a physical location (not distributed), or chassis. The patch ports on this router are known as l3gateway ports in the ovn-southbound database (`ovn-sbdb`). For example in the following `ovnkube-trace` snippet the gateway routers are prefixed with `GR_`:
++
+[source,terminal]
+----
+ * Logical datapaths:
+  *     "GR_ip-10-0-156-14.us-east-2.compute.internal" (649fab4d-b6a8-4290-ae44-da42d3416bf6) [ingress]
+  *     "GR_ip-10-0-147-232.us-east-2.compute.internal" (89e6d20d-7ad2-4e1a-84fd-4e98a6e2e23a) [ingress]
+  *     "GR_ip-10-0-149-128.us-east-2.compute.internal" (9f27517e-2440-402e-a14c-763bf7b07209) [ingress]
+  *     "GR_ip-10-0-206-73.us-east-2.compute.internal" (b6626ff9-8d06-4dc5-9665-ed34815f732d) [ingress]
+  *     "ovn_cluster_router" (d5fc1961-658e-4b2d-acf2-70c3b4584b8c) [ingress]
+  *     "GR_ip-10-0-140-161.us-east-2.compute.internal" (e44aa51c-21ff-4cda-b748-89a6f18a6da6) [ingress]
+  * Logical flow: table=1 (lr_in_lookup_neighbor), priority=0, match=(1), actions=(reg9[2] = 1; next;)
+----
+
+Distributed logical routers:: Distributed logical routers and the logical switches behind them, to which virtual machines and containers attach, effectively reside on each hypervisor. For example in the following `ovnkube-trace` snippet the distributed logical router is identified as `ovn_cluster_router`.
++
+[source,terminal]
+----
+*     "GR_ip-10-0-206-73.us-east-2.compute.internal" (b6626ff9-8d06-4dc5-9665-ed34815f732d) [ingress]
+  *     "ovn_cluster_router" (d5fc1961-658e-4b2d-acf2-70c3b4584b8c) [ingress]
+  *     "GR_ip-10-0-140-161.us-east-2.compute.internal" (e44aa51c-21ff-4cda-b748-89a6f18a6da6) [ingress]
+----
+Join local switch:: Join local switches are used to connect the distributed router and gateway routers. It reduces the number of IP addresses needed on the distributed router. For example in the following `ovnkube-trace` snippet the join logical switch is identified as `join`.
++
+[source,terminal]
+----
+* Logical datapaths:
+  *     "ip-10-0-149-128.us-east-2.compute.internal" (0255bd08-b156-4b50-a078-a5d6b06208b9) [ingress]
+  *     "ip-10-0-140-161.us-east-2.compute.internal" (12d1e5b1-5346-43b5-b723-6fa4c9b07445) [ingress]
+  *     "ip-10-0-206-73.us-east-2.compute.internal" (398ca3e4-e70b-4632-af4e-64f88e17c133) [ingress]
+  *     "join" (5db354c9-37b6-4139-beab-25fa009bcb16) [ingress]
+  *     "ext_ip-10-0-156-14.us-east-2.compute.internal" (70b4031c-bc50-47e6-bcc5-94aae13c4ca5) [ingress]
+  *     "ext_ip-10-0-206-73.us-east-2.compute.internal" (98366c43-7d8d-4b6d-8d83-0a64a68175a2) [ingress]
+----
+
+Logical switches with patch ports:: Logical switches with patch ports are used to virtualize the network stack. They connect remote logical ports through tunnels.
```

Comment:
To find the 'patch' ports:
 ./network-tools ovn-db-run-command ovn-sbctl find Port_Binding type=patch